src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets

```diff
@@ -1,59 +1,38 @@
 <?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="GenerateResourcesCode" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll" />
 
-  <!-- #################################### -->
-  <!-- Generate Resources code -->
-  <!-- #################################### -->
   <PropertyGroup>
-    <_ResPath>$(ToolsDir)ResourceGenerator.exe</_ResPath>
-    <_JSon2TxtPath>$(ToolsDir)JSon2Txt.exe</_JSon2TxtPath>
-    <_ResInputFileFullPath>$(ResInputPath)\resources.resjson</_ResInputFileFullPath>
-    <_ResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(BaseIntermediateOutputPath)$(Configuration)\$(AssemblyName).SR.cs</_ResOutputFileFullPath>
-    <_ResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(BaseIntermediateOutputPath)$(Configuration)\$(AssemblyName).SR.vb</_ResOutputFileFullPath>
-    <_ResTxtFullPath>$(BaseIntermediateOutputPath)$(Configuration)\$(AssemblyName).restext</_ResTxtFullPath>
-    <_ResIntermediateOutputPath>$(BaseIntermediateOutputPath)$(Configuration)\Res\</_ResIntermediateOutputPath>
-    <_ResPriIntermediateOutputFilePath>$(_ResIntermediateOutputPath)$(AssemblyName).pri</_ResPriIntermediateOutputFilePath>
-    <_ResJsonIntermediateOutputFilePath>$(_ResIntermediateOutputPath)$(AssemblyName).resources.resjson</_ResJsonIntermediateOutputFilePath>
+    <ResourcesSourceOutputDirectory Condition="'$(ResourcesSourceOutputDirectory)' == ''">$(MSBuildProjectDirectory)\Resources</ResourcesSourceOutputDirectory>
+    <StringResourcesPath Condition="'$(StringResourcesPath)'=='' And Exists('$(ResourcesSourceOutputDirectory)\Strings.resx')">$(ResourcesSourceOutputDirectory)\Strings.resx</StringResourcesPath>
+    <ResourceSRPath Condition="'$(ResourceSRPath)'==''">$(ResourcesSourceOutputDirectory)\SR.cs</ResourceSRPath>
+    <IntermediateResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)SR.cs</IntermediateResOutputFileFullPath>
+    <IntermediateResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(IntermediateOutputPath)SR.vb</IntermediateResOutputFileFullPath>
```

Comment:
Actually you caught some issue which ResourceSRPath should check for cs or vb. so I need to add extra line there
but in general I think this is the only file we need to support VB. in VB projects we may manually need to add compile item to the SR.vb (I am trying to not create more extra targets file for VB as we don't have many VB projects)